distribution.pm

```diff
@@ -171,6 +172,74 @@ sub script_sudo {
     return;
 }
 
+=head2 script_output
+
+    script_output($script, [ $wait, type_command => ? ]) 
+
+Execute $script on the SUT and return the data written to STDOUT by
+$script. See script_output in the testapi.
+
+You may be able to avoid overriding this function by setting
+$serial_term_prompt.
+
+=cut
+sub script_output {
+    my ($self, $script, $wait, %args) = @_;
+    my $marker = testapi::hashed_string("SO$script");
+    # 80 is approximate quantity of chars typed during 'curl' approach
+    # if script length is lower there is no point to proceed with more complex solution
+    $args{type_command} //= length($script) < 80;
+    my $script_path = "/tmp/script$marker.sh";
+
+    if (testapi::is_serial_terminal) {
+        my $cat = "cat - > $script_path; echo $marker-\$?-";
+        testapi::wait_serial($self->{serial_term_prompt}, undef, 0, no_regex => 1);
+        testapi::type_string($cat . "\n");
+        testapi::wait_serial("$cat", undef, 0, no_regex => 1);
+        testapi::type_string($script);
+        testapi::type_string("\n", terminate_with => 'EOT');
+        testapi::wait_serial("$marker-0-");
+    }
+    elsif ($args{type_command}) {
+        my $cat = "cat - > $script_path;\n";
+        testapi::type_string($cat);
+        testapi::type_string($script . "\n");
+        testapi::send_key('ctrl-d');
+    }
+    else {
+        open my $fh, ">", 'current_script' or croak("Could not open file. $!");
+        print $fh $script;
+        close $fh;
+        testapi::assert_script_run("curl -f -v " . testapi::autoinst_url("/current_script") . " > $script_path");
+        testapi::script_run "clear";
+    }
+
+    # Surround the actual script output with special markers so that we can
+    # unambiguously separate the expected output from other content that we
+    # might encounter on the serial device depending on how it is used in the
+    # SUT
+    my $shell_cmd = testapi::is_serial_terminal ? 'bash -oe pipefail' : 'bash -eox pipefail';
```

Comment:
this is right way :
`my $shell_cmd = testapi::is_serial_terminal() ? 'bash -oe pipefail' : 'bash -eox pipefail';`
currently it is not compiling with your version of code

Reply:
Works on my computer... but I will change it :-p